### ISSUE TITLE
Multiple Expansions not expanding

### ISSUE THREAD

**Issue by pzl** [fiskhest/sxhkd-helper-menu#4]
`sxhkd` supports multiple expansions, like the following:

```
super + {_,shift + }{1-9,0}
  bspc {desktop -f, node -d} ^{1-9,10}
```

which, as a first "intermediate" expansion might resemble the following pair:

```
super + {1-9,0}
  bspc desktop -f ^{1-9,10}

super + shift + {1-9,0}
  bspc node -d ^{1-9,10}
```

Which this project is then typically capable of handling. But the first form, having compounding expansions doesn't seem to be supported. This project _does_ seem like it performs the first-order expansion currently, but not any subsequent ones

**Comment by fiskhest** [comment]
Nice catch! I'm not entirely certain I'll be able to add support for that use case within a reasonable time frame but I will give it a try.

**Comment by Hoboneer** [comment]
Hi, I'm writing a parser for sxhkd config files.

It does proper parsing (tokenising and all), so it handles these cases.  It's still WIP, but it's mostly complete now.  It's written in Python (>=3.7) and has no dependencies, but is not yet on PyPI.

It's at https://github.com/Hoboneer/sxhkd-parser

**Comment by fiskhest** [comment]
Would anyone know if sxhkd supports more than two compounded expansions per line?

I finally seem to have found a way to solve this and fixup the codebase somewhat in the process, but the new solution only supports two expansions per line. I honestly can't see the use case for doing three being supported, but I did overlook this feature for two expansions so I'm not going to drop it without asking.

**Comment by Hoboneer** [comment]
I recall using 3 sequences of the form `{s1,s2,...,sn}` in the same keybind---it worked, but I didn't keep it.  I imagine it won't be all that common in the wild though.

**Comment by fiskhest** [comment]
Thanks for the reply @Hoboneer 

When reading sxhkd's man-page I stumbled upon the wording `The hotkey and the command may contain sequences of the form {STRING_1,...,STRING_N}.`, since the wording is plural sequences my understanding would be that it is supported using `n` sequences per line, matching your findings.

I will take a stab at supporting that too, but I have yet to find one sxhkdrc-file in the wild where someone is doing more than two expansions per line. If a quick fix cannot be applied, I might just drop support for more than two compounding expansions and clarify in the readme.

**Comment by fiskhest** [comment]
I have released 0.4.0 which addresses this issue. Using two or more `{seq,...}` per line is now supported, and the codebase is a little bit more maintainable now 😅 

Example:
```
# Go to {desktop, node} {1-9,10}
super + {_,shift + }{1-9,0}
  bspc {desktop -f, node -d} ^{1-9,10}
```

Produces:
```
Go to desktop 1    super + 1           bspc desktop -f ^1   
Go to desktop 2    super + 2           bspc desktop -f ^2   
Go to desktop 3    super + 3           bspc desktop -f ^3   
Go to desktop 4    super + 4           bspc desktop -f ^4   
Go to desktop 5    super + 5           bspc desktop -f ^5   
Go to desktop 6    super + 6           bspc desktop -f ^6   
Go to desktop 7    super + 7           bspc desktop -f ^7   
Go to desktop 8    super + 8           bspc desktop -f ^8   
Go to desktop 9    super + 9           bspc desktop -f ^9   
Go to desktop 10   super + 0           bspc desktop -f ^10  
Go to node 1       super + shift + 1   bspc node -d ^1      
Go to node 2       super + shift + 2   bspc node -d ^2      
Go to node 3       super + shift + 3   bspc node -d ^3      
Go to node 4       super + shift + 4   bspc node -d ^4      
Go to node 5       super + shift + 5   bspc node -d ^5      
Go to node 6       super + shift + 6   bspc node -d ^6      
Go to node 7       super + shift + 7   bspc node -d ^7      
Go to node 8       super + shift + 8   bspc node -d ^8      
Go to node 9       super + shift + 9   bspc node -d ^9      
Go to node 10      super + shift + 0   bspc node -d ^10     
```

Thank you @pzl for finding and creating the issue!

**Comment by fiskhest** [comment]
Actually noticed a few more bugs when testing around on my own configuration and when trying to upload those fixes I also ran into a couple of publishing issues. This is finally fixed in `v0.4.3`.
